pkgs/applications/science/math/eukleides/default.nix

```
@@ -1,26 +1,43 @@
 { lib, stdenv, fetchurl, bison, flex, texinfo, readline, texLive }:
 
-let
-  name    = "eukleides";
+lib.fix (eukleides: stdenv.mkDerivation rec {
+  pname = "eukleides";
   version = "1.5.4";
-in
-stdenv.mkDerivation {
-  name = "${name}-${version}";
 
   src = fetchurl {
-    url = "http://www.eukleides.org/files/${name}-${version}.tar.bz2";
+    url = "http://www.eukleides.org/files/${pname}-${version}.tar.bz2";
     sha256 = "0s8cyh75hdj89v6kpm3z24i48yzpkr8qf0cwxbs9ijxj1i38ki0q";
   };
 
-  buildInputs = [bison flex texinfo readline texLive];
+  nativeBuildInputs = [ bison flex ];
 
-  preConfigure = "sed -i 's/ginstall-info/install-info/g' doc/Makefile";
-  installPhase = "mkdir -p $out/bin ; make PREFIX=$out install";
+  buildInputs = [ texinfo readline texLive ];
+
+  preConfigure = ''
+    sed -i 's/ginstall-info/install-info/g' doc/Makefile
+    sed -i '/mktexlsr 2> \/dev\/null/d' Makefile
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    make PREFIX=$out install
+    mv $out/share/texmf/tex $tex
+    ln -s "$tex" "$out/share/texmf/tex"
+
+    runHook postInstall
+  '';
+
+  outputs = [ "out" "tex" ];
```

```suggestion
  outputs = [ "out" "doc" "tex" ];
```
because I mentioned `$doc` in my `substituteInPlace` suggestion. Adding the `doc` output is just finessing the derivation more than anything.

---

Done